src/test/java/software/amazon/awssdk/crt/test/CredentialsProviderTest.java

```diff
@@ -151,4 +157,34 @@ public Credentials getCredentials() {
             fail(ex.getMessage());
         }
     }
+
+    @Test
+    public void testCreateDestroyProfile_ValidProfile() throws IOException {
+        Path credsPath = Files.createTempFile("testCreateDestroyProfile_ValidProfile_creds_", "");
+        Files.write(credsPath, Arrays.asList("[default]"));
+
+        try {
+            ProfileCredentialsProvider.ProfileCredentialsProviderBuilder builder =
+                    new ProfileCredentialsProvider.ProfileCredentialsProviderBuilder()
+                    .withCredentialsFileNameOverride(credsPath.toString());
+
+            try (ProfileCredentialsProvider provider = builder.build()) {
+                assertNotNull(provider);
+                assertTrue(provider.getNativeHandle() != 0);
+            }
+        } finally {
+            Files.deleteIfExists(credsPath);
+        }
+    }
+
+    @Test(expected = CrtRuntimeException.class)
+    public void testCreateDestroyProfile_InvalidProfile() {
```

Comment:
Why is this test going to throw an exception?

Reply:
Because the config/credentials files will not contain a default profile, triggering [this error condition](https://github.com/awslabs/aws-c-auth/blob/a5dbb0cb7fccb6405b16245f4be7cbd2225496bd/source/credentials_provider_profile.c#L392-L399). I will add comments more clearly indicating why the exception is expected.

Reply:
But you're not overriding the default which means its going to the local users files which may very well have those profiles (like mine when I ran it, for example).

Reply:
Oh good point. I'll manually create empty files without profiles to verify this error condition.